packages/schemas/src/foundations/jsonb-types/verification-records.ts

```diff
@@ -0,0 +1,46 @@
+import { socialUserInfoGuard } from '@logto/connector-kit';
+import { z } from 'zod';
+
+export const defaultVerificationRecordDataGuard = z.object({});
+
+export const emailVerificationRecordDataGuard = z.object({
+  email: z.string(),
+});
+
+export type EmailVerificationRecordData = z.infer<typeof emailVerificationRecordDataGuard>;
+
+export const phoneVerificationRecordDataGuard = z.object({
+  phone: z.string(),
+});
+
+export type PhoneVerificationRecordData = z.infer<typeof phoneVerificationRecordDataGuard>;
+
+export const socialVerificationRecordDataGuard = z.object({
+  connectorId: z.string(),
+  socialUserInfo: socialUserInfoGuard.optional(),
+});
+
+export type SocialVerificationRecordData = z.infer<typeof socialVerificationRecordDataGuard>;
+
+export const verificationRecordDataGuard = z.union([
+  defaultVerificationRecordDataGuard,
+  emailVerificationRecordDataGuard,
+  phoneVerificationRecordDataGuard,
+  socialVerificationRecordDataGuard,
+]);
```

Comment:
Can we reuse the JSON data guard defined in the verification record class?

Reply:
Removed this typing.